### PR TITLE
Reposition popover on scroll & remove tooltip delayed from screen

### LINKED ISSUE #259: Reposition popover on scroll and remove tooltip when scrolling
### Describe the feature
Currently on touch devices, when a user is scrolling, the tooltip is following when scrolling, it doesn't reposition and the tolltip is not disappearing.

A suggestion for this is to reposition the popover when scrolling so it always anchors to its anchor element.
Also on touch devices, you touch the screen constantly, so before removing the tooltip the user should scroll a certain distance. From playing around with it, 150 pixels feels good.

### Added value
Better user experience for pad and phone users.

### PR DESCRIPTION
**Pull request by danielkaxis**
### Describe your changes

To give better user experience for phone and pad users, reposition popover on scroll event.
Also remove the tooltip when user, on a touch device, scrolled more than 150 pixels.

### Issue ticket number and link

- Fixes #259 

### Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have verified that the code builds perfectly fine on my local system
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have verified that my code follows the style already available in the repository
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by danielkaxis** [comment]
I'm wondering about respositioning the popover on scroll. It has a slight studder when scrolling and I'm not sure if we can fix it with the current implementation of how popover positions itself to the anchor element.

**Comment by lekoaf** [comment]
> Also remove the tooltip when user, on a touch device, scrolled more than 150 pixels.

Do tooltips even show up on a touch device? I thought they only showed on hover...?

**Comment by danielkaxis** [comment]
> > Also remove the tooltip when user, on a touch device, scrolled more than 150 pixels.
> 
> Do tooltips even show up on a touch device? I thought they only showed on hover...?

Yes, when you touch the element.

File: packages/core/src/Tooltip/utils/useTouchScrollDistance.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  const [origin, setOrigin] = useState<TouchList | null>(null)
+  const [touches, setTouches] = useState<TouchList | null>(null)
+  /**
+   * The distance between touch origin and touch current for both
+   * x-axis and y-axis
+   */
+  const [touchScrollDistance, setTouchScrollDistance] = useState({ x: 0, y: 0 })
+
+  useEffect(() => {
+    const touchstartHandler = (event: TouchEvent) => {
```
Comment (boilund) [inline packages/core/src/Tooltip/utils/useTouchScrollDistance.ts:13]: touch**S**tartHandler?

File: packages/core/src/Tooltip/utils/useTouchScrollDistance.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      if (origin === null) {
+        setOrigin(event.touches)
+      }
+    }
+
+    const touchmoveHandler = (event: TouchEvent) =>
```
Comment (boilund) [inline packages/core/src/Tooltip/utils/useTouchScrollDistance.ts:19]: touch**M**oveHandler

File: packages/core/src/Tooltip/utils/useTouchScrollDistance.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      setTouches(event.changedTouches)
+
+    const touchendHandler = (event: TouchEvent) => {
```
Comment (boilund) [inline packages/core/src/Tooltip/utils/useTouchScrollDistance.ts:22]: touch**E**ndHandler?

File: packages/core/src/Tooltip/utils/useTouchScrollDistance.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      if (event.touches.length === 0) {
+        setOrigin(null)
+        setTouches(null)
+        setTouchScrollDistance({ x: 0, y: 0 })
+      }
+    }
+
+    const touchcancelHandler = () => {
```
Comment (boilund) [inline packages/core/src/Tooltip/utils/useTouchScrollDistance.ts:30]: touch**C**ancelHandler?

**Comment by boilund** [comment]
Can you add tests as well?

File: packages/core/src/Tooltip/utils/useTouchScrollDistance.ts (hunk 4 above)
Comment (danielkaxis) [reply packages/core/src/Tooltip/utils/useTouchScrollDistance.ts]: Done

File: packages/core/src/Tooltip/utils/useTouchScrollDistance.ts (hunk 3 above)
Comment (danielkaxis) [reply packages/core/src/Tooltip/utils/useTouchScrollDistance.ts]: Done

File: packages/core/src/Tooltip/utils/useTouchScrollDistance.ts (hunk 2 above)
Comment (danielkaxis) [reply packages/core/src/Tooltip/utils/useTouchScrollDistance.ts]: Done

File: packages/core/src/Tooltip/utils/useTouchScrollDistance.ts (hunk 1 above)
Comment (danielkaxis) [reply packages/core/src/Tooltip/utils/useTouchScrollDistance.ts]: Done

**Comment by danielkaxis** [comment]
> Can you add tests as well?

Will look into it

**Pull request merged**